### PR TITLE
Docker auth config

### PR DESCRIPTION
**Pull request by DavidTanner**
window

[Dockerode](https://github.com/apocas/dockerode/blob/ad8f7a083458036d155658a45cbf3ad453605a14/lib/docker.js#L90) wants the auth settings in `opts.authconfig`

### CODE REVIEW

**Review by RyanRatcliff — APPROVED** [review]

File: src/docker.ts [hunk 1]
```diff
@@ -79,8 +79,10 @@ const buildAuthForDocker = () => {
   if (dockerUser && dockerPass) {
     logger.debug(`Pulling image as ${dockerUser}`);
     return {
-      username: dockerUser,
-      password: dockerPass
+      authConfig: {
```
Comment (smolster) [inline src/docker.ts:82]: Based on the code you linked [here](https://lifeomic.slack.com/archives/C01948JLJG7/p1623693030119600?thread_ts=1623686917.104300&cid=C01948JLJG7), I think this should be all-lowercase `authconfig`

**Review by smolster — DISMISSED** [review]

File: src/docker.ts (hunk 1 above)
Comment (DavidTanner) [reply src/docker.ts]: The one time I didn't CP...

**Review by IevgenRagulin — APPROVED** [review]

**Comment by DavidTanner** [comment]
Looks like 16 needs to be added, and 10 removed.

**Review by RyanRatcliff — DISMISSED** [review]

**Comment by RyanRatcliff** [comment]
Sorry, should have mentioned these had tests backing them up.  You'll want to update the response body of that as well.

**Comment by tqhoughton** [comment]
nodejs 10.x test is failing with:

```
  docker › pullImage › will provide credentials from env variables when pulling an image from docker hub

  Rejected promise returned by test. Reason:

  Error {
    json: null,
    reason: 'server error',
    statusCode: 500,
    message: '(HTTP code 500) server error - Head https://registry-1.docker.io/v2/library/alpine/manifests/3.5: unauthorized: incorrect username or password ',
  }

  › node_modules/docker-modem/lib/modem.js:257:17
  › IncomingMessage.<anonymous> (node_modules/docker-modem/lib/modem.js:284:9)
```

**Review by smolster — APPROVED** [review]

**Review by RyanRatcliff — APPROVED** [review]

**Review by tqhoughton — APPROVED** [review]

**Pull request merged**